contrib/contrib.cmake

```
@@ -107,6 +107,17 @@ if (NOT LINALG_LIBS)
   else (NOT DEFINED LINALG_LIBS)
     message (STATUS "OpenBLAS not found. We'll build it from source.")
 
+    # OpenBLAS will build without Fortran, but it will silently neglect to build vital functions
+    include( CheckLanguage )
```

does this just print the warning? I think we want to fail completely here, right?

---

It does print the warning but I don't see any reason for it to fail there.

---

the point of this PR is to make explicit the problems that occur when we build openblas without a fortran compiler. if we don't have fortran, but are trying to build openblas, we want to fail loudly - one warning message behind several screens of openblas compile barf won't help us.

---

if it doesn't fail, won't the build just fail later on when trying to link? Generally, we'd like for things to fail at the cause, not the result.

---

it doesn't help us to continue building under those conditions, right?

---

to be clear, when I say "fail here", I mean fail when we don't have a fortran compiler, not at this specific line.